src/rollup/RollupCreator.sol

```diff
@@ -45,24 +44,27 @@ contract RollupCreator is Ownable {
         address eigenDARollupManager;
     }
 
-    BridgeCreator public bridgeCreator;
-    IOneStepProofEntry public osp;
-    IChallengeManager public challengeManagerTemplate;
-    IRollupAdmin public rollupAdminLogic;
-    IRollupUser public rollupUserLogic;
-    IUpgradeExecutor public upgradeExecutorLogic;
+    modifier onlyUnfrozen() {
+        require(!deploymentFrozen, "Deployment no longer permitted from this RollupCreator");
+        _;
+    }
 
-    address public validatorUtils;
-    address public validatorWalletCreator;
+    BridgeCreator public immutable bridgeCreator;
+    IOneStepProofEntry public immutable osp;
+    IChallengeManager public immutable challengeManagerTemplate;
+    IRollupAdmin public immutable rollupAdminLogic;
+    IRollupUser public immutable rollupUserLogic;
+    IUpgradeExecutor public immutable upgradeExecutorLogic;
 
-    DeployHelper public l2FactoriesDeployer;
+    address public immutable validatorUtils;
+    address public immutable validatorWalletCreator;
 
-    constructor() Ownable() {}
+    DeployHelper public immutable l2FactoriesDeployer;
 
-    // creator receives back excess fees (for deploying L2 factories) so it can refund the caller
-    receive() external payable {}
+    bool public deploymentFrozen;
+    string public version;
 
-    function setTemplates(
+    constructor(
```

Comment:
biggest concern with removing `setTemplates` is that this new patten breaks the deploy script flows: https://github.com/Layr-Labs/nitro-contracts/blob/eigenda-v2.1.0/scripts/local-deployment/deployCreator.ts#L50-L66